site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1922,6 +1922,53 @@ public function getPostsForThread($current_user, $thread_id){
       return $result_rows;
     }
 
+    public function mergeThread($parent_thread_id, $child_thread_id, &$message){
+        try{
+            $this->course_db->beginTransaction();
+
+            $this->course_db->query("SELECT 1 FROM threads WHERE id = ? and merged_id = -1 and deleted = false", array($child_thread_id));
+            $result_rows = $this->course_db->rows();
+            if(count($result_rows)<1) {
+                $message = "Can't find child thread";
+                $this->course_db->rollback();
+                return false;
+            }
+            $this->course_db->query("SELECT 1 FROM threads WHERE id = ? and merged_id = -1 and deleted = false", array($parent_thread_id));
+            $result_rows = $this->course_db->rows();
+            if(count($result_rows)<1) {
+                $message = "Can't find parent thread";
+                $this->course_db->rollback();
+                return false;
+            }
```

Comment:
Lines 1929-1942 has duplicate code. It might be worth creating a function that returns a boolean to whether or not a non merged thread exists.